src/main.js

```diff
@@ -33,50 +33,56 @@ const mostCommentedMovies = movies.slice().sort(compareCommentsAmount);
 
 
 const profileRatingViewComponent = new ProfileRatingView(alreadyWatchedMovies.length);
-render(headerElement, profileRatingViewComponent.element, RenderPosition.BEFOREEND);
+render(headerElement, profileRatingViewComponent, RenderPosition.BEFOREEND);
 
 const menuComponent = new MenuView(watchlistMovies, alreadyWatchedMovies, favoriteMovies);
-render(mainElement, menuComponent.element, RenderPosition.AFTERBEGIN);
+render(mainElement, menuComponent, RenderPosition.AFTERBEGIN);
 
 const filmsCatalogComponent = new FilmsCatalogView();
-render(mainElement, filmsCatalogComponent.element, RenderPosition.BEFOREEND);
+render(mainElement, filmsCatalogComponent, RenderPosition.BEFOREEND);
 
 const footerStatisticsViewComponent = new FooterStatisticsView(movies);
-render(footerStatisticElement, footerStatisticsViewComponent.element, RenderPosition.AFTERBEGIN);
+render(footerStatisticElement, footerStatisticsViewComponent, RenderPosition.AFTERBEGIN);
 
 const sortingViewComponent =  new SortingView();
-render(menuComponent.element, sortingViewComponent.element, RenderPosition.AFTEREND);
+render(menuComponent, sortingViewComponent, RenderPosition.AFTEREND);
 
 const filmCardsContainerElement = filmsCatalogComponent.element.querySelector('.films-list__container');
 
-bodyELement.addEventListener('click', (evt) => {
-  evt.preventDefault();
+const renderPopup = (currentMovieIndex) => {
 
-  if (evt.target.closest('.film-card__link')) {
-    bodyELement.classList.add('hide-overflow');
+  bodyElement.classList.add('hide-overflow');
 
-    const isCurrentMovie = (movie) => (
-      movie.id === Number(evt.target.closest('.film-card__link').querySelector('.film-card__input').value)
-    );
+  const popupComponent = new PopupView(movies[movies.findIndex(currentMovieIndex)]);
+  render(footerElement, popupComponent, RenderPosition.AFTEREND);
+  const filmCommentsContainer = popupComponent.element.querySelector('.film-details__comments-list');
 
-    const popupComponent = new PopupView(movies[movies.findIndex(isCurrentMovie)]);
-    render(footerElement, popupComponent.element, RenderPosition.AFTEREND);
-    const filmCommentsContainer = document.querySelector('.film-details__comments-list');
+  const currentComments = movies[movies.findIndex(currentMovieIndex)].comments;
 
-    for (let k = 0; k <= currentComments.length - 1; k++) {
-      const filmCommentViewComponent = new FilmCommentView(currentComments[k]);
-      render(filmCommentsContainer, filmCommentViewComponent.element, RenderPosition.AFTERBEGIN);
-    }
+  for (let k = 0; k <= currentComments.length - 1; k++) {
+    const filmCommentViewComponent = new FilmCommentView(currentComments[k]);
+    render(filmCommentsContainer, filmCommentViewComponent, RenderPosition.AFTERBEGIN);
+  }
```

Comment:
Циклы где нам не нужны индесксы лучше делать через `for..of`
```suggestion
  for (let currentComment of currentComments) {
    const filmCommentViewComponent = new FilmCommentView(currentComment);
    render(filmCommentsContainer, filmCommentViewComponent, RenderPosition.AFTERBEGIN);
  }
```

Reply:
или метод `forEach`
```suggestion
  currentComments.forEach((comment) => {
    const filmCommentViewComponent = new FilmCommentView(comment);
    render(filmCommentsContainer, filmCommentViewComponent, RenderPosition.AFTERBEGIN);
  })
```